flutter/lib/src/sentry_asset_bundle.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:flutter/material.dart';
+import 'package:sentry/sentry.dart';
+
+typedef _Parser<T> = Future<T> Function(String value);
+
+/// An [AssetBundle] which creates automatic performance traces for loading
+/// assets.
+///
+/// You can wrap other [AssetBundle]s in it:
+/// ```dart
+/// SentryAssetBundle(bundle: someOtherAssetBundle)
+/// ```
+/// If you're not providing any [AssetBundle], it falls back to the [rootBundle].
+///
+/// If you want to use the [SentryAssetBundle] by default you can achieve this
+/// with the following code:
+/// ```dart
+/// DefaultAssetBundle(
+///   bundle: SentryAssetBundle(),
+///   child: MaterialApp(
+///     home: MyScaffold(),
+///   ),
+/// );
+/// ```
+/// [Image.asset], for example, will then use [SentryAssetBundle].
+class SentryAssetBundle implements AssetBundle {
+  SentryAssetBundle({Hub? hub, AssetBundle? bundle})
+      : _hub = hub ?? HubAdapter(),
+        _bundle = bundle ?? rootBundle;
+
+  final Hub _hub;
+  final AssetBundle _bundle;
+
+  @override
+  Future<ByteData> load(String key) async {
+    final span = _hub.getSpan()?.startChild(
+          'file.read',
+          description: 'AssetBundle.load(key=$key)',
+        );
+
+    try {
+      final data = await _bundle.load(key);
+      await span?.finish(status: SpanStatus.ok());
+      return data;
+    } catch (e) {
+      await span?.finish(status: SpanStatus.internalError());
+      rethrow;
+    }
+  }
+
+  /// Does not create a span. Sometimes [CachingAssetBundle] can throw errors
+  /// which are outside the current zone. This is not easy to handle and can
+  /// result in corrupt spans.
+  @override
+  Future<T> loadStructuredData<T>(String key, _Parser<T> parser) =>
```

Comment:
I couldn't get this to reliably work in the error case, so no support for now. Though, I believe this is the most interesting method because of the parsing :(

Reply:
if we just wrap on a try-catch like the others, isn't that enough for now?

Reply:
No, it can throw errors which somehow escape the async context and thus never reaches the try-catch. It's very weird. The Sentry root zone integration catches it though, so that's nice.

Reply:
Okay, I've finally got it to work 🎉